packages/java-edition/src/mcfunction/checker/index.ts

```diff
@@ -97,7 +98,108 @@ const itemPredicate: core.SyncChecker<ItemPredicateNode> = (node, ctx) => {
 		for (const allOfTest of anyOfTest.children) {
 			for (const test of allOfTest.children) {
 				const key = core.ResourceLocationNode.toString(test.key, 'full')
-				if (ComponentTestExactNode.is(test) && test.value) {
+				// count is a special case that's only valid in item predicate arguments, not json
+				// note: basically all errors checked here are otherwise accepted by vanilla, but it's good to report them
+				if (key === 'minecraft:count') {
+					if (ComponentTestExistsNode.is(test)) {
+						ctx.err.report(
+							localize('mcfunction.checker.item-predicate.useless-count-check'),
+							test.range,
+							core.ErrorSeverity.Warning,
+						)
+					} else if (
+						(ComponentTestExactNode.is(test) || ComponentTestSubpredicateNode.is(test))
+						&& test.value
+					) {
```

Comment:
The whole body of this if statement should be checked by mcdoc ideally. But I understand you did not want to do that in this PR. We should not have these very specific error messages for every use case.